core/sys/darwin/mach_darwin.odin

```diff
@@ -1,29 +1,514 @@
 package darwin
 
-foreign import pthread "system:System.framework"
+foreign import mach "system:System.framework"
 
 import "core:c"
+import "base:intrinsics"
 
-// NOTE(tetra): Unclear whether these should be aligned 16 or not.
-// However all other sync primitives are aligned for robustness.
-// I cannot currently align these though.
-// See core/sys/unix/pthread_linux.odin/pthread_t.
-task_t :: distinct u64
-semaphore_t :: distinct u64
+kern_return_t :: distinct c.int
 
-kern_return_t :: distinct u64
-thread_act_t :: distinct u64
+mach_port_t :: distinct c.uint
+vm_map_t :: mach_port_t
+mem_entry_name_port_t :: mach_port_t
+ipc_space_t :: mach_port_t
+thread_t :: mach_port_t
+task_t :: mach_port_t
+semaphore_t :: mach_port_t
+
+vm_size_t :: distinct c.uintptr_t
+
+vm_address_t :: vm_offset_t
+vm_offset_t :: distinct c.uintptr_t
+
+// NOTE(beau): typedefed to int in the original headers
+boolean_t :: b32
+
+vm_prot_t :: distinct c.int
+
+vm_inherit_t :: distinct c.uint
+
+mach_port_name_t :: distinct c.uint
 
 @(default_calling_convention="c")
-foreign pthread {
-	mach_task_self :: proc() -> task_t ---
+foreign mach {
+	mach_task_self :: proc() -> mach_port_t ---
+
+	semaphore_create :: proc(task: task_t, semaphore: ^semaphore_t, policy, value: c.int) -> Kern_Return ---
+	semaphore_destroy :: proc(task: task_t, semaphore: semaphore_t) -> Kern_Return ---
+
+	semaphore_signal :: proc(semaphore: semaphore_t) -> Kern_Return ---
+	semaphore_signal_all :: proc(semaphore: semaphore_t) -> Kern_Return ---
+	semaphore_signal_thread :: proc(semaphore: semaphore_t, thread: thread_t) -> Kern_Return ---
+
+	semaphore_wait :: proc(semaphore: semaphore_t) -> Kern_Return ---
+
+	vm_allocate :: proc (target_task : vm_map_t, address: ^vm_address_t, size: vm_size_t, flags: VM_Flags) -> Kern_Return ---
+
+	vm_deallocate :: proc(target_task: vm_map_t, address: vm_address_t, size: vm_size_t) -> Kern_Return ---
+
+	vm_map :: proc (
```

Comment:
The way of aligning these isn't really how it's normally done in the codebase.
This is usually the way:
```odin
proc(
    target_task: vm_map_t,
    address:     ^vm_address_t,
   ...
```

also it's a nit but could you make sure there is no space between `proc` and `(` anywhere.